crowd_evacuation/civilian_agent.py

```diff
@@ -13,12 +14,37 @@ def __init__(self, unique_id, model, known_exits):
         self._known_exits = known_exits
         self._strategy = "random"
         self._willingness_to_follow_steward = random.uniform(0, 1)
-        self._speed = random.uniform(3, 10)
         self._age = random.randrange(15, 65)
-        self._gender = random.choice(["M", "F"])
-        self._size = random.uniform(40, 100)
+        # self._gender = random.choice(["M", "F"])  not used yet. Maybe for later. Gender can affect speed (see papers)
+        self._weight = random.uniform(40, 100)
         self._goal = None
         self._interacted_with = []
+        self._exit_point = (None, None)  # a tuple that stores the exit of the agent
+        self._visual_range = self.calculate_visual_rage(self._age)
+        self._speed = self.calculate_speed(self._visual_range, self._weight)
+
+    def calculate_visual_rage(self, age):
+        """
+        :param age:
+        :return: the visual range of the agent based on age
+        """
+        if age <= 45:
+            visual_range = random.randrange(5, 7)  # randomness accounts for individual variation
+        else:
+            visual_range = random.randrange(4, 6)
+        return visual_range
+
+    def calculate_speed(self, visual_range, weight):
+        """
+        :param visual_range:
+        :param weight:
+        :return: speed according to age and size. Speed varies between 3 and 7 m/s as per previous studies
+        """
+        if weight > 70:
+            speed = visual_range - random.randrange(1, 3)   # weight has negative penalty on the speed
+        else:
+            speed = visual_range    # else, the speed is the same as the visual range
+        return speed
```

Comment:
I think it's good that the speed depends on the vision range, but maybe limit it to 4m/s max.